src/theory/strings/word.cpp

```diff
@@ -76,6 +76,29 @@ size_t Word::getLength(TNode x)
   return 0;
 }
 
+std::vector<Node> Word::getChars(TNode x)
+{
+  Kind k = x.getKind();
+  if (k == CONST_STRING)
+  {
+    std::vector<Node> ret;
+    NodeManager* nm = NodeManager::currentNM();
+    std::vector<unsigned> ccVec;
+    const std::vector<unsigned>& cvec = x.getConst<String>().getVec();
+    for (unsigned i = 0, size = cvec.size(); i < size; i++)
+    {
+      ccVec.clear();
+      ccVec.insert(ccVec.end(), cvec.begin() + i, cvec.begin() + i + 1);
+      Node ch = nm->mkConst(String(ccVec));
+      ret.push_back(ch);
+    }
```

Comment:
It seems to me as if this could be simplified to:

```suggestion
    for (unsigned chVal : x.getConst<String>().getVec())
    {
      ccVec.clear();
      ccVec.push_back(chVal);
      Node ch = nm->mkConst(String(ccVec));
      ret.push_back(ch);
    }
```